tgui/packages/tgui/interfaces/RiotGear.js

```diff
@@ -0,0 +1,211 @@
+    cooldown,
+  } = data;
+
+  const [nameModal, displayNameModal] = useLocalState(context, 'nameIsOnFile', '');
+  const [printModal, displayPrintModal] = useLocalState(context, 'printIsOnFile', '');
+
+  return (
+    <Window
+      title={`Armoury Authorisation`}
+      width={800}
+      height={280}
+    >
+      <Window.Content>
+        {(!hasID && !authed) && (
+          <Modal
+            fontSize="20px"
+            mr={2}
+            p={3}>
+            <Box>
+              No ID Given!
+            </Box>
+          </Modal>
+        )}
+        {(!hasAccess && !(!hasID) && !authed) && (
+          <Modal
+            fontSize="20px"
+            mr={2}
+            p={3}>
+            <Box>
+              Insufficient Access Level!
+            </Box>
+          </Modal>
+        )}
+        {(!(!mismatchedID) && !(!hasAccess) && !(!hasID) && !authed) && (
+          <Modal
+            fontSize="20px"
+            mr={2}
+            p={3}>
+            <Box>
+              ID Registration Does Not Match User!
+            </Box>
+          </Modal>
+        )}
+        {(!(!mismatchedAppearance) && !mismatchedID && !(!hasAccess) && !(!hasID) && !authed) && (
+          <Modal
+            fontSize="20px"
+            mr={2}
+            p={3}>
+            <Box>
+              User Appearance Does Not Match User Voiceprint!
+            </Box>
+          </Modal>
+        )}
+        {(!(!cooldown) && !mismatchedAppearance && !mismatchedID && !(!hasAccess) && !(!hasID)) && (
+          <Modal
+            fontSize="20px"
+            mr={2}
+            p={3}>
+            <Box>
+              <TimeDisplay value={cooldown} timing={"true"} format={formatTime} /> Before Any Commands May Be Accepted Again.
+            </Box>
+          </Modal>
+        )}
+        {(!maxSecAccess && !cooldown && !(!authed)) && (
+          <Modal
+            fontSize="20px"
+            mr={2}
+            p={3}>
+            <Box>
+              Armoury Access Has Been Authorised!
+            </Box>
+          </Modal>
+        )}
+        {nameModal && (
+          <Modal
+            fontSize="20px"
+            mr={2}
+            p={3}>
+            <Box>
+              Authorisation Already Issued By User!
+            </Box>
+          </Modal>
+        )}
+        {printModal && (
+          <Modal
+            fontSize="20px"
+            mr={2}
+            p={3}>
+            <Box>
+              User Fingerprint ID Already On File!
+            </Box>
+          </Modal>
+        )}
+
+        <Section>
+          <Flex>
+            <Flex.Item width="50%">
+              <LabeledControls.Item>
+                <Button
+                  width={30}
+                  height={3.5}
+                  textAlign={"center"}
+                  fontSize={2}
+                  color={existingAuthorisation ? 'blue' : 'green'}
+                  disabled={authed ? true : false}
+                  onClick={() => (maxSecAccess
```

Comment:
I do not understand this ternary, so this logic should probably happen at least partially on the backend, or at least be an if chain or something.

Reply:
Forgive me for the rather poor JS code, still rather new to it. The logic there first checks if the user has both a fingerprint and their name on record or has neither on record, then if so, then gives the option to authorise or repeal based on whether they already have an authorisation on file. If not, it will display an error modal stating that either their fingerprint is on file, or their name is on file, based on the `nameOnFile` variable.

Reply:
Okay, it would probably make more sense to figure this out on the backend and then only send things like `canAuth` and so on, instead of sending fingerprint data to the frontend to do the checks there. Failing that, this could at least be an if else structure so it's somewhat readable.